packages/grid/_modules_/grid/models/api/gridRowApi.ts

```diff
@@ -5,10 +5,10 @@ import { GridRowModel, GridRowId, GridRowModelUpdate } from '../gridRows';
  */
 export interface GridRowApi {
   /**
-   * Get the full set of rows as [[Rows]].
-   * @returns [[Rows]]
+   * Get the full set of rows as [[Map<GridRowId, GridRowModel>]].
+   * @returns [[Map<GridRowId, GridRowModel>]]
    */
```

Comment:
I might be missing something. Wasn't this public method meant to return all the rows as they exist in the grid (e.g. if updated with the streaming API)? Shouldn't developers expect a flat array? I mean, why do developers need to know about the id? Not a major concern, more a question.

Reply:
Actually, this method could return a flat array. Although, changing it also changes the `rows` prop of the return of `getFilterModelParams` [here](https://github.com/mui-org/material-ui-x/blob/ba5640495504090541b6fc9bb22eff7da036f02d/packages/grid/_modules_/grid/hooks/features/filter/useGridFilter.ts#L44). Then, we would have `rows` as a flat array and `visibleRows` as a map, not very consistently.

Reply:
There are 2 use cases here. 
- The grid uses the api internally and for perf reason, we want to get the id with rows so we don't have to run getRowId for every rows and state update. 
- We can consider adding a method, that would get the values of the map so we get a flat array of rows for the users or we can wait for an issue?

Reply:
No preferences. I personally think that the current state of the PR is good enough. I was wondering about either there is a potential for improvement or not.